src/Bicep.Core/Parsing/ParamsParser.cs

```diff
@@ -0,0 +1,1139 @@
+
+                // if skipped node is returned above, the newline is not consumed
+                // if newline token is returned, we must not expect another (could be a beginning of a declaration)
+                if (declarationOrToken is ITopLevelDeclarationSyntax)
+                {
+                    // declarations must be followed by a newline or the file must end
+                    var newLine = this.WithRecoveryNullable(this.NewLineOrEof, RecoveryFlags.ConsumeTerminator, TokenType.NewLine);
+                    if (newLine != null)
+                    {
+                        declarationsOrTokens.Add(newLine);
+                    }
+                }
+            }
+
+            var endOfFile = reader.Read();
+
+            return new ProgramSyntax(declarationsOrTokens, endOfFile, this.lexerDiagnostics);
+        }
+
+        public SyntaxBase Statement() =>
+            this.WithRecovery(
+                () =>
+                {
+
+                    Token current = reader.Peek();
+
+                    return current.Type switch
+                    {
+                        TokenType.Identifier => current.Text switch
+                        {
+                            LanguageConstants.UsingKeyword => this.UsingDeclaration(),
+                            LanguageConstants.ParameterKeyword => this.ParameterAssignment(),
+                            _ => throw new ExpectedTokenException(current, b => b.UnrecognizedDeclaration()),
+                        },
+                        TokenType.NewLine => this.NewLine(),
+
+                        _ => throw new ExpectedTokenException(current, b => b.UnrecognizedDeclaration()),
+                    };
+                },
+                RecoveryFlags.None,
+                TokenType.NewLine);
+
+        private UsingDeclarationSyntax UsingDeclaration()
+        {
+            var keyword = ExpectKeyword(LanguageConstants.UsingKeyword);
+            var path = this.WithRecovery(
+                () => ThrowIfSkipped(this.InterpolableString, b => b.ExpectedModulePathString()),
+                GetSuppressionFlag(keyword),
+                TokenType.Assignment, TokenType.NewLine);
+            
+            return new UsingDeclarationSyntax(keyword, path);
+        
+        }
+        
+        private SyntaxBase ParameterAssignment()
+        {
+            var keyword = ExpectKeyword(LanguageConstants.ParameterKeyword);
+            var name = this.IdentifierWithRecovery(b => b.ExpectedParameterIdentifier(), RecoveryFlags.None, TokenType.Identifier, TokenType.NewLine);
+            var assignment = this.WithRecovery(this.Assignment, GetSuppressionFlag(name), TokenType.NewLine);
+            var value = this.WithRecovery(() => this.Expression(ExpressionFlags.AllowComplexLiterals), GetSuppressionFlag(assignment), TokenType.NewLine);
+
+            return new ParameterAssignmentSyntax(keyword, name, assignment, value);
+        }
+
+        private Token Assignment()
```

Comment:
The methods below are identical to the ones in `Parser.cs`. Consider extract them to a base class, e.g., `BaseParser` and let `ParamsParser` and `Parser` inherit the base class to share the method definitions. An alternative approach is just to merge `ParamsParser` into `Parser` and change `Program` to `ParamsProgram`.

It's up to you to decide which option is better, and I'm open to do the refactoring in a separate PR when adding integration tests for the parameters file parser. We can also have a discussion with the team about this.